action.yml

```diff
@@ -134,39 +134,93 @@ runs:
         git add -f ${{ github.workspace }}/dist/\*
+        echo "Changed files:"
+        echo "$(git diff-index --cached --name-only HEAD)"
         if [ -n "$(git diff-index --cached --name-only HEAD)" ]; then
-          git commit -m "${{ inputs.commitMessage }}" || echo "Commit failed"
-          # Attempt to pull and resolve conflicts
-          git pull origin ${{ github.ref_name }} || {
-            echo "Rebase failed, force pushing changes."
-            git push --force origin HEAD:${{ github.ref_name }}
-            exit 0
-          }
-          git push origin HEAD:${{ github.ref_name }}
+          node -e "
+            const fs = require('fs');
+            const path = require('path');
+            const github = require('@actions/github');
+            const manifestPath = path.relative('.', '${{ inputs.manifestPath }}');
+
+            async function pushChanges() {
+              const octokit = github.getOctokit(process.env.GITHUB_TOKEN);
+
+              const context = github.context;
+              const owner = context.repo.owner;
+              const repo = context.repo.repo;
+              const ref = 'heads/${{ github.ref_name }}';
+
+              try {
+                const currentCommit = await octokit.rest.git.getRef({
+                  owner,
+                  repo,
+                  ref
+                });
+
+                const newTree = await octokit.rest.git.createTree({
+                  owner,
+                  repo,
+                  base_tree: currentCommit.data.object.sha,
+                  tree: [
+                    {
+                      path: manifestPath,
+                      mode: '100644',
+                      type: 'blob',
+                      content: fs.readFileSync(manifestPath, 'utf8'),
+                    },
+                    ...fs.readdirSync('${{ github.workspace }}/dist').map((file) => {
+                        const relativePath = path.relative('.', '${{ github.workspace }}/dist/' + file);
+                        return {
+                          path: relativePath,
+                          mode: '100644',
+                          type: 'blob',
+                          content: fs.readFileSync(
+                          '${{ github.workspace }}/dist/' + file,
+                          'utf-8',
+                        ),
+                      };
+                    }),
+                  ],
+                });
+
+                const newCommit = await octokit.rest.git.createCommit({
+                  owner,
+                  repo,
+                  message: '${{ inputs.commitMessage }}',
+                  tree: newTree.data.sha,
+                  parents: [currentCommit.data.object.sha]
+                });
+
+                await octokit.rest.git.updateRef({
+                  owner,
+                  repo,
+                  ref,
+                  sha: newCommit.data.sha,
+                  force: true
```

Comment:
If you have to force it in CI you;re doing something wrong

Reply:
It's a security because if two build run concurrently, even by pulling before one can push during the pull of the other, I have that case before. And no merge can be done because binary files are pushed.